src/phase0/state_transition/mod.rs

```diff
@@ -44,6 +45,8 @@ pub enum Error {
     Overflow,
     #[error("{0}")]
     InvalidOperation(InvalidOperation),
+    #[error("overflow")]
```

Comment:
```suggestion
    #[error("invalid slot provided")]
```

Reply:
want to extend this enum variant w/ data so the error is more descriptive?

```rust
#[error("an invalid transition to a past slot {requested} from slot {current}")]
TransitionToPreviousSlot { current: Slot, requested: Slot}
```

Reply:
ok